### ISSUE TITLE
[v2 BUG]: Cross entropy loss is too big

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#845]
`chemprop.nn.MulticlassClassificationFFN` sets `n_tasks = n_tasks * n_classes`. This is a problem because then the default task weights is `task_weights = torch.ones(n_tasks)`. `F.cross_entropy` in `CrossEntropyLoss` will reduce the `n_tasks * n_classes` predicted values to `n_tasks` loss values, but then `L = L  * self.task_weights.view(1, -1)` will broadcast the `n_tasks` loss values to `n_tasks * n_classes`, effectively multiplying the loss value by `n_classes` when we use `L.sum()`.

This doesn't break anything so I'll plan to fix it when we take another look at loss functions and metrics.

**Comment by davidegraff** [comment]
specifically, the problem starts here:

https://github.com/chemprop/chemprop/blob/cc5b3c11e0aedff3abadff43b9c0860fa07a4370/chemprop/nn/predictors.py#L256-L283

and continues to here:

https://github.com/chemprop/chemprop/blob/cc5b3c11e0aedff3abadff43b9c0860fa07a4370/chemprop/nn/predictors.py#L108-L131

IMO it will probably be easier to fix this by abstracting the multiclass FFN into its own `Predictor` subclass rather than abusing the `_FFNPredictorBase` class like we do currently.
